decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class Decorator < Nameable
+  def initialize(name = '')
+    super()
+    @nameable = name
+  end
```

Comment:
- [ ] **[OPTIONAL]** It would be great if you can rename the parameter to be more descriptive as this class doesn't initialize with a name but with a `nameable` object! 🤓 🔧 

```suggestion
  def initialize(nameable)
    super()
    @nameable = nameable
  end
```